code/iomethods.py

```diff
@@ -5,39 +5,116 @@
+import netCDF4
 
 import utils
 import variables
 
+def _get_fileList_for_binaries(dirPath, startTime, endTime):
+    '''
+        Purpose:: There are two kinds of files in the MERG directory. One has files in binary form, and the other in
+                  netCDF. This function is to get only the files that are in between the startTime and endTime that are
+                  in binary format.
+        Input:: dirPath: Path to where the binary MERG files are.
+                startTime: The cutoff for the earliest files you want
+                endTime: The cutoff for the latest files you want
+        Output:: A list of a path to files that are between startTime and endTime
+        Assumptions:: startTime and endTime are in the format YYYYMMDDHHSS
+                      The filename always ends with -pixel
+                      The date is included in the file name, with the format YYYYMMDDHH
+    '''
+    fileString = os.path.join(dirPath, '*-pixel')
+    fileList = glob.glob(fileString)
+    fileList.sort()
+
+    newFileList = []
+
+    start = datetime.strptime(startTime, '%Y%m%d%H%S')  # Convert start and end times to datetimes to compare with
+    end = datetime.strptime(endTime, '%Y%m%d%H%S')      # the current file's time
+
+    for file in fileList:
+        dateFromFileName = file[5:15]
+        dateAsDateTime = datetime.strptime(dateFromFileName, '%Y%m%d%H')
+
+        if start <= end and start <= dateAsDateTime <= end:
+            newFileList.append(file)
+        elif start <= dateAsDateTime or dateAsDateTime <= end:
+            newFileList.append(file)
+
+    return newFileList
+
+
+def _check_pattern(dirPath, startTime, endTime, tdelta, tRes, flag, filelist):
+    '''
+        Purpose:: refactored pattern recognition to be more modular and usable by other functions that
+                    that wants to obtain the startTime, startFile, endTime, endFile
+    '''
+    
+    startFile = ''
+    endFile = ''
+    currFile = ''
+    hasDelimiter = False
+    startTimeInFile = ''
+    endTimeInFile = ''
+    currTimeInFile = ''
+    tokenCounter = 0
+    filenamePattern = ''
+
+    if flag is True:
+        for eachPart in re.split(r'[_,-,.,/]', re.split(r'.nc', os.path.basename(filelist[0]))[0]):
+            tokenCounter += 1
+            if tokenCounter == 1:
+                filenamePattern += eachPart
+            if eachPart.isdigit():
+                if len(eachPart) >= 6:
+                    hasDelimiter = True
+                    startTimeInFile += eachPart + '*'
+                elif eachPart in startTime:
+                    startTimeInFile += eachPart + '*'
+    
+        if hasDelimiter is False:
+            fileDate = int(re.search(r'\d+', re.split(r'.nc', os.path.basename(filelist[0]))[0]).group())
+            filenamePattern = re.split(str(fileDate), os.path.basename(filelist[0]))[0]
+    else:
+        for eachPart in re.split(r'[_,-,.,/]', re.split(os.path.basename(filelist[0]))[0]):
```

Comment:
re.split takes two parameters. re.split(os.path.basename(filelist[0]))[0]) is missing a parameter